R/filter_parameters.R

```diff
@@ -80,11 +80,11 @@ filter_by_join <- function(x, parameters = NULL, nm = "") {
   }
   extra_names <- setdiff(filter_names, param_names)
   if (length(extra_names) > 0) {
-    msg <- paste0(
-      "There are unneeded columns in `parameters` that were ignored: ",
-      paste0("'", extra_names, "'", collapse = ", ")
-    )
-    rlang::warn(msg)
+    cli_warn(c(
+      "{qty(extra_names)} The column{?s} {.var {extra_names}} passed in \\
+       {.arg parameters} {?is/are} unneeded and will be ignored."
+    ))
```

Comment:
it'd be nice to pass the call through from `filter_parameters()`

Reply:
Agreed, though unfortunately `rlang::warn()` (via `cli_warn`) doesn't accept a `call` argument in the way that `rlang::abort()` (via `cli_abort()`) does. :/

Reply:
Oh, I see! Do you know if that's intentional?

Reply:
I'm not sure what the logic is there. I was wrong here, though: looks like there is indeed something happening the ellipses:

``` r
rlang::warn("boop")
#> Warning: boop
rlang::warn("boop", call = "bop")
#> Warning in "bop": boop
```

<sup>Created on 2022-12-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Will update in a moment!

Reply:
Oh, but it looks like the machinery in that argument doesn't work quite the same.

With `caller_env()` in this warnings `call`, we see:

```
Warning message:
In filter_parameters(svm_reg_results, parameters = tibble::tibble(`%^*#` = 1,  :
  The column `soup` passed in `parameters` is not needed and will be ignored.
```